f3read.c

```diff
@@ -379,6 +386,12 @@ static void iterate_files(const char *path, const long *files,
 		tot_overwritten += stats.secs_overwritten;
 		tot_size += stats.bytes_read;
 		and_read_all = and_read_all && stats.read_all;
+		if (stats.secs_corrupted && delete) {
+			const char *filename;
+			char *full_fn = full_fn_from_number(&filename, path, *files);
+			if (unlink(full_fn))
+				err(errno, "Can't remove file %s\n", full_fn);
+		}
```

Comment:
1. Add `assert(full_fn);` immediatly after line `char *full_fn = full_fn_from_number(...`
2. Add `free(full_fn);` immediatly after line `err(errno, "Can't remove file %s\n", full_fn);` Notice that it won't belong to the body of the statement `if (unlink(full_fn))`.